### PR TITLE
Refactor azure queue scaler

### PR DESCRIPTION
**Pull request by rickbrouwer**
- Refactor the scaler to match the new metadata guidelines.
- In #5875 the contributor indicated that he needed help with making the tests. These were not there yet, so I added them during the refactoring

### Checklist

- [X] I have verified that my change is according to the [deprecations & breaking changes policy](https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md)
- [X] Tests have been added
- [X] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

Related: [5797](https://github.com/kedacore/keda/pull/5797)

### CODE REVIEW

**Comment by rickbrouwer** [comment]
Since Microsoft is actually the maintainer of Azure scalers I don't know if this PR is allowed.
I did my best, but a good review is desired!

**Comment by wozniakjan** [comment]
> Since Microsoft is actually the maintainer of Azure scalers I don't know if this PR is allowed.

It is allowed and also it is much appreciated :heart:
Can I kindly ask you @SpiritZhou to also take a look for any Microsoft-specific requirements?

**Comment by rickbrouwer** [comment]
Hmm....i see some errors in the azure_event_hub_dapr_wi_test.
Something is going wrong there while deleting the template: `helper.go:609: Deleting template: triggerAuthTemplate`

File: pkg/scalers/azure_queue_scaler.go
Comment (SpiritZhou) [inline pkg/scalers/azure_queue_scaler.go:60]: The Validate() is not needed if there is nothing to check.

File: pkg/scalers/azure_queue_scaler.go
Comment (SpiritZhou) [inline pkg/scalers/azure_queue_scaler.go:97]: Validate() is not needed to be called. It will be called during TypedConfig().

File: pkg/scalers/azure_queue_scaler.go
Comment (SpiritZhou) [inline pkg/scalers/azure_queue_scaler.go:102]: I think this check can be moved to Validate(). Perhaps it can be refactored later, as there are other Azure scalers that have not been refactored yet and still refer to it.

File: pkg/scalers/azure_queue_scaler.go
Comment (rickbrouwer) [reply pkg/scalers/azure_queue_scaler.go]: solved

File: pkg/scalers/azure_queue_scaler.go
Comment (rickbrouwer) [reply pkg/scalers/azure_queue_scaler.go:93]: Please let we refactor this later together with the other Azure scalers

**Comment by rickbrouwer** [comment]
Let's wait to run the e2e for this PR until #6271 is merged and this branch is updated

**Review by SpiritZhou — APPROVED** [review]

**Review by SpiritZhou — APPROVED** [review]

**Review by JorTurFer — APPROVED** [review]

**Review by wozniakjan — APPROVED** [review]
lgtm, thank you!

**Pull request merged**
